core/src/test/scala/integration/kafka/api/BaseQuotaTest.scala

```diff
@@ -136,6 +141,28 @@ abstract class BaseQuotaTest extends IntegrationTestHarness {
     assertTrue("Should have been throttled", consumerThrottleMetric.value > 0)
   }
 
+  @Test
+  def testThrottledRequest() {
+
+    overrideQuotas(Long.MaxValue, Long.MaxValue, 0.1)
+    waitForQuotaUpdate(Long.MaxValue, Long.MaxValue, 0.1)
+
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singleton(topic1))
+    val endTimeMs = System.currentTimeMillis + 10000
+    var throttled = false
+    while (!throttled && System.currentTimeMillis < endTimeMs) {
+      consumer.poll(100)
```

Comment:
Hmm, when there is no data, the consumer will wait for the timeout. So, not sure if this is enough to trigger the throttling. We probably need to either set a low maxWait in consumer config and set the quota to be really low.